controlplane/api/v1alpha1/rke2controlplane_types.go

```diff
@@ -71,6 +72,11 @@ type RKE2ControlPlaneSpec struct {
 	// the registration type is "address". Its for scenarios where a load-balancer or VIP is used.
 	// +optional
 	RegistrationAddress string `json:"registrationAddress,omitempty"`
+
+	// The RolloutStrategy to use to replace control plane machines with new ones.
+	// +optional
+	// +kubebuilder:default={type: "RollingUpdate", rollingUpdate: {maxSurge: 1}}
```

Comment:
I was thinking that we shouldn't adopt this default like KCP to make CAPRKE2 be more GitOps friendly. 

Problem with using defaults is that it causes the original spec to differ from the one that was applied, which can then cause a GitOps implementation to think there is drift.

Reply:
Should we instead apply the default as part of the reconciliation loop?

Reply:
Or we could probably make any change around defaults in another PR, so its applied across all the APIs

Reply:
Can we do it in v1alpha2?

Reply:
Created #196 to track changes for v1alpha2.